src/indicators/money_flow_index.rs

```diff
@@ -0,0 +1,235 @@
+/// ![Money Flow Index, stockcharts] (https://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:money_flow_index_mfi)
+
+#[derive(Debug, Clone)]
+pub struct MoneyFlowIndex {
+    n: u32,
+    mf_vec: VecDeque<f64>,
+    prev_tp: f64,
+}
+
+impl MoneyFlowIndex {
+    pub fn new(n: u32) -> Result<Self> {
+        match n {
+            0 => Err(Error::from_kind(ErrorKind::InvalidParameter)),
+            _ => {
+                let indicator = Self {
+                    n: n,
+                    mf_vec: VecDeque::with_capacity(n as usize + 1),
+                    prev_tp: 0.0,
+                };
+                Ok(indicator)
+            }
+        }
+    }
+}
+
+impl Next<f64> for MoneyFlowIndex {
+    type Output = f64;
+
+    fn next(&mut self, mf: f64) -> f64 {
+        if (self.mf_vec.len() as u32) < self.n {
+            self.mf_vec.push_back(mf);
+        } else {
+            self.mf_vec.pop_front();
+            self.mf_vec.push_back(mf);
+        }
+
+        if self.mf_vec.len() as u32 == self.n {
+            let pmf: f64 = self.mf_vec.iter().filter(|&i| *i >= 0.0).sum();
+            let nmf: f64 = self.mf_vec.iter().filter(|&i| *i < 0.0).sum();
+            ((pmf / (pmf + nmf.abs())) * 100.0)
+        } else {
+            0.0
+        }
+    }
+}
+
+impl<'a, T: High + Low + Close + Volume> Next<&'a T> for MoneyFlowIndex {
+    type Output = f64;
+
+    fn next(&mut self, input: &'a T) -> f64 {
+        // self.next((input.high(), input.low(), input.close(), input.volume()))
+
+        let high = input.high();
+        let low = input.low();
+        let close = input.close();
+        let volume = input.volume();
+
+        let tp = (high + low + close) / 3.0;
+        let rmf = tp * volume;
+        let mf = if tp >= self.prev_tp {
+            rmf
+        } else {
+            (rmf * -1.0)
+        };
+
+        self.prev_tp = tp;
+
+        if (self.mf_vec.len() as u32) < self.n {
+            self.mf_vec.push_back(mf);
+        } else {
+            self.mf_vec.pop_front();
+            self.mf_vec.push_back(mf);
+        }
+
+        if self.mf_vec.len() as u32 == self.n {
+            let pmf: f64 = self.mf_vec.iter().filter(|&i| *i >= 0.0).sum();
+            let nmf: f64 = self.mf_vec.iter().filter(|&i| *i < 0.0).sum();
+
+            ((pmf / (pmf + nmf.abs())) * 100.0)
+        } else {
+            0.0
+        }
+    }
+}
+
+impl Default for MoneyFlowIndex {
+    fn default() -> Self {
+        Self::new(14).unwrap()
+    }
+}
+
+impl fmt::Display for MoneyFlowIndex {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "MFI({})", self.n)
+    }
+}
+
+impl Reset for MoneyFlowIndex {
+    fn reset(&mut self) {
+        self.mf_vec.clear();
+        self.prev_tp = 0.0;
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test_helper::*;
+
+    test_indicator!(MoneyFlowIndex);
```

Comment:
This probably partially makes sense but will break, if we remove `Next<64>`.
I need to split this test helper into fewer smaller ones.
For now you can remove it.

Reply:
For time being added dummy Next<f64> implementation to test MFI.
Will remove Next<f64> as helper is modified.